guides/hosting/gitlab.md

```diff
@@ -0,0 +1,145 @@
+
+The test stage of the pipeline will either pass (each element of the array returned a healthy exit code) or it will fail (one of the elements returned an error).
+
+If it passes, then the pipeline will move onto the second job we defined here which [we must name](https://docs.gitlab.com/ee/ci/yaml/#pages) "pages". This is a special job just for deploying content to your gitlab pages site! This one is executed after tests have passed because we specified that it should occur in the "deploy" stage. It again runs the commands in `script` (this time building the docs), but this time we tell it to preserve the path `public` (where we stashed the docs) as an artifact of the job.
+
+The result of naming this job `pages` and putting our docs in the `public` directory and specifying it as an `artifact` is that GitLab will deploy the site in that directory to the default URL `https://<YOUR-GITLAB-USERNAME>.gitlab.io/<YOUR-REPOSITORY-NAME>`.
+
+The `before_script` and `cache` keys in the file are for running the same script in every job (`shards install`) and for hanging onto the files that were created (`cache`). They're not necessary if your shard doesn't have any dependencies.
+
+If you commit the above file to your project and push, you'll trigger your first run of the new pipeline.
+
+```bash
+$ git add -A && git commit -am 'Add .gitlab-ci.yml' && git push origin master
+```
+
+### Some Badges
+
+While that pipeline is running, let's attach some badges to the project to show off our docs and the (hopefully) successful state of our pipeline. (You might want to read the [badges docs](https://gitlab.com/help/user/project/badges).)
+
+A badge is just a link with an image. So let's create a link to our pipeline and fetch a badge image from the [Gitlab Pipeline Badges API](https://docs.gitlab.com/ee/user/project/pipelines/settings.html#pipeline-badges).
+
+In the _Badges_ section of the _General_ settings, we'll first add a release badge. The link is: `https://gitlab.com/<YOUR-GITLAB-USERNAME>/<YOUR-REPOSITORY-NAME>/pipelines` and the _Badge Image URL_ is: `https://gitlab.com/<YOUR-GITLAB-USERNAME>/<YOUR-REPOSITORY-NAME>/badges/master/pipeline.svg`.
+
+And now if the pipleline has finished we'll have docs and we can link to them with a generic badge from `shields.io`.
+
+- Link: `https://<YOUR-GITLAB-USERNAME>.gitlab.io/<YOUR-REPOSITORY-NAME>`
+- Image: `https://img.shields.io/badge/docs-available-brightgreen.svg`
+
+
+## Releases
+
+A release is just a special commit in your history with a name (see [tagging](https://git-scm.com/book/en/v2/Git-Basics-Tagging)).
+
+According to [the Crystal Shards README](https://gitlab.com/crystal-lang/shards/blob/master/README.md),
```

Comment:
Think this needs to be: https://github.com/crystal-lang/shards/blob/master/README.md